src/shared/components/ResourceEditor/useEditorTooltip.tsx

```diff
@@ -0,0 +1,405 @@
+import * as React from 'react';
+import CodeMirror from 'codemirror';
+import clsx from 'clsx';
+import { useNexusContext } from '@bbp/react-nexus';
+import { useSelector } from 'react-redux';
+import {
+  TEditorPopoverResolvedData,
+  editorLinkResolutionHandler,
+  getTokenAndPosAt,
+  mayBeResolvableLink,
+} from './editorUtils';
+import { TDELink } from '../../store/reducers/data-explorer';
+import { RootState } from '../../store/reducers';
+import useResolutionActions from './useResolutionActions';
+
+const downloadImg = require('../../images/DownloadingLoop.svg');
+
+export const CODEMIRROR_HOVER_CLASS = 'CodeMirror-hover-tooltip';
+export const CODEMIRROR_LINK_CLASS = 'fusion-resource-link';
+type TTooltipCreator = Pick<
```

Comment:
Isn't type `TTooltipCreator` the same as `TReturnedResolvedData`? Maybe we can reuse it instead of creating a new one?